### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by AVyha**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,29 @@
...
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(victim):
+        if isinstance(victim, Herbivore) and not victim.hidden:
+            victim.health -= 50
+        if victim.health <= 0:
+            Animal.alive.pop(Animal.alive.index(victim))
```
Comment (wQuelS) [inline app/main.py:29]: consider using remove instead of pop. index() is not so reliable

File: app/main.py [hunk 2]
```diff
@@ -1 +1,29 @@
...
+
+
+class Herbivore(Animal):
+    def hide(self):
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(victim):
```
Comment (wQuelS) [inline app/main.py:25]: add type annotation to function and arg

**Review by wQuelS — CHANGES_REQUESTED** [review]

**Review by wQuelS — APPROVED** [review]
looks good imo

File: app/main.py [hunk 3]
```diff
@@ -1 +1,29 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name, health: int = 100):
```
Comment (Y-Havryliv) [inline app/main.py:4]: add type annotations

File: app/main.py [hunk 4]
```diff
@@ -1 +1,29 @@
...
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{" \
```
Comment (Y-Havryliv) [inline app/main.py:11]: rewrite it, use `( )` for multiline strings

**Review by Y-Havryliv — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1 +1,27 @@
...
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(victim: Herbivore) -> None:
+        if isinstance(victim, Herbivore) and not victim.hidden:
+            victim.health -= 50
+        if victim.health <= 0:
+            victim.alive.remove(victim)
```
Comment (Y-Havryliv) [inline app/main.py:27]: it's better to get access to class attributes using class name(`Animal`)

**Review by Y-Havryliv — CHANGES_REQUESTED** [review]

**Review by Y-Havryliv — APPROVED** [review]
Nice!

**Pull request closed without merging**